crates/cubecl-std/src/tensor/layout/linear.rs

```diff
@@ -0,0 +1,125 @@
+#[derive(CubeType, CubeLaunch, Clone)]
+pub enum LinearLayout {
+    /// Input is contiguous, no mapping
+    Plain { len: u32 },
+    /// Strided tensor, i.e. freshly allocated but not permuted
+    Strided(StridedLayout),
+    /// Permuted layout, tracks the entire shape/strides and not just the last dim
+    Permuted(PermutedLayout),
+}
+
+impl<'a, R: Runtime> LinearLayoutArgs<'a, R> {
+    pub fn from_shape_strides(
+        client: &ComputeClient<R::Server, R::Channel>,
+        shape: &[usize],
+        strides: &[usize],
+        line_size: &'a u8,
+    ) -> Self {
+        let rank = shape.len();
+        if rank == 1 || is_contiguous(shape, strides) {
+            let len = shape.iter().product::<usize>();
+            let len = len / *line_size as usize;
+            Self::Plain {
+                len: ScalarArg::new(len as u32),
+            }
+        } else if is_contiguous_pitched(shape, strides) {
+            Self::Strided(StridedLayoutLaunch::from_shape_strides(
+                client, shape, strides, line_size,
+            ))
+        } else {
+            Self::Permuted(PermutedLayoutLaunch::from_shape_strides(
+                client, shape, strides, line_size,
+            ))
+        }
+    }
+
+    pub fn from_handle(
+        client: &ComputeClient<R::Server, R::Channel>,
+        handle: &TensorHandleRef<'a, R>,
+        line_size: &'a u8,
+    ) -> Self {
+        Self::from_shape_strides(client, handle.shape, handle.strides, line_size)
+    }
+}
+
+#[cube]
+impl Layout for LinearLayout {
+    type Coordinates = Coords1d;
+
+    fn to_linear_pos(this: &Self, pos: Self::Coordinates) -> u32 {
+        match this {
+            LinearLayout::Plain { .. } => pos,
+            LinearLayout::Strided(strided_layout) => strided_layout.to_linear_pos(pos),
+            LinearLayout::Permuted(permuted_layout) => permuted_layout.to_linear_pos(pos),
+        }
+    }
+
+    fn to_linear_pos_checked(this: &Self, pos: Self::Coordinates) -> (u32, bool) {
+        match this {
+            LinearLayout::Plain { len } => {
+                let idx = this.to_linear_pos(pos);
+                let in_bounds = pos < *len;
+                (idx, in_bounds)
+            }
+            LinearLayout::Strided(strided_layout) => strided_layout.to_linear_pos_checked(pos),
+            LinearLayout::Permuted(permuted_layout) => permuted_layout.to_linear_pos_checked(pos),
+        }
+    }
+
+    fn shape(this: &Self) -> Self::Coordinates {
+        match this {
+            LinearLayout::Plain { len } => *len,
+            LinearLayout::Strided(strided_layout) => strided_layout.shape(),
+            LinearLayout::Permuted(permuted_layout) => permuted_layout.shape(),
+        }
+    }
+}
+
+virtual_layout!(LinearLayout, LinearLayoutExpand);
+
+pub type LinearTensorView<E, IO = Read> = TensorViewTyped<E, LinearLayout, IO>;
+pub type LinearTensorViewLaunch<'a, R> = TensorViewTypedLaunch<'a, LinearLayout, R>;
```

Comment:
We should add docs if those types are supposed to be used often. Also I'm not a fan of the naming, should we drop the `View` postfix? Only `LinearTensor` is probably enough info. What do you think?

Reply:
I did the rename 👍